server/src/main/java/org/elasticsearch/action/support/CancellableFanOut.java

```diff
@@ -94,7 +95,20 @@ public final void run(@Nullable Task task, Iterator<Item> itemsIterator, ActionL
                 final ActionListener<ItemResponse> itemResponseListener = ActionListener.notifyOnce(new ActionListener<>() {
                     @Override
                     public void onResponse(ItemResponse itemResponse) {
-                        onItemResponse(item, itemResponse);
+                        try {
+                            onItemResponse(item, itemResponse);
+                        } catch (Exception e) {
```

Comment:
I don't have much experience with this code, but why do we want to enforce no exception, rather than calling onItemFailure with the exception?

Reply:
Calling `onItemFailure` in that case is effectively a double-completion for that item (we already called `onItemResponse`) which tends to be a little trappy. If the implementation wants to feed exceptions into its `onItemFailure` method then it can do so itself.

Reply:
Sounds good, thanks.